business/definitionService.js

```diff
@@ -176,22 +179,76 @@ class DefinitionService {
     )
   }
 
-  /**
-   * Compute facets related to an entity and a curation of that entity. This is essentially a light
-   * weight pre-compute so we know what groupings to use in the real summarization.
-   *
-   * @param {EntitySpec} coordinates - The entity for which we are looking for a curation
-   * @param {*} raw - the set of raw tool ouptuts related to the idenified entity
-   * @param {Definition)} [curation] - an actual curation object to apply.
-   * @returns {Facets} The `facets` portion of the component's described definition
-   */
-  async _computeFacets(coordinates, raw, curation) {
-    // TODO we might be able to a more effecient computation here since all we need are the facets
-    // for example, some tool outputs just will never have facets to there is no point in summarizing them.
-    const summarized = await this.summaryService.summarizeFacets(coordinates, raw)
-    const aggregated = await this.aggregationService.process(coordinates, summarized)
-    const definition = await this.curationService.apply(coordinates, curation, aggregated)
-    return get(definition, 'described.facets')
+  // ensure all the right facet information has been computed and added to the given definition
+  _ensureFacets(definition) {
+    if (!definition.files) return
+    const facetFiles = this._computeFacetFiles([...definition.files], definition.described.facets)
+    for (const facet in facetFiles)
+      setIfValue(definition, `licensed.facets.${facet}`, this._summarizeFacetInfo(facet, facetFiles[facet]))
+  }
+
+  // figure out which files are in which facets
+  _computeFacetFiles(files, facets = {}) {
+    const facetList = Object.getOwnPropertyNames(facets)
+    remove(facetList, 'core')
+    if (facetList.length === 0) return { core: files }
+    const result = { core: [...files] }
+    for (const facet in facetList) {
+      const facetKey = facetList[facet]
+      const filters = facets[facetKey]
+      if (!filters || filters.length === 0) break
+      result[facetKey] = files.filter(file => filters.some(filter => minimatch(file.path, filter)))
+      pullAllWith(result.core, result[facetKey], isEqual)
+    }
+    return result
+  }
+
+  // create the data object for the identified facet containing the given files. Also destructively brand
+  // the individual file objects with the facet
+  _summarizeFacetInfo(facet, facetFiles) {
+    if (!facetFiles || facetFiles.length === 0) return null
+    const attributions = new Set()
+    const licenseExpressions = new Set()
+    let unknownParties = 0
+    let unknownLicenses = 0
+    // accummulate all the licenses and attributions, and count anything that's missing
+    for (let file of facetFiles) {
+      file.license ? licenseExpressions.add(file.license) : unknownLicenses++
+      const statements = this._simplifyAttributions(file.attributions)
+      statements ? addArrayToSet(statements, attributions) : unknownParties++
+      if (facet !== 'core') {
```

Comment:
should we bother checking for 'core' here even though we removed 'core' in `_computeFacetFiles` just before this method?